crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -290,6 +291,42 @@ static_assert(not is_subtype_of(Literal[0], AlwaysTruthy))
 
 static_assert(not is_subtype_of(str, AlwaysTruthy))
 static_assert(not is_subtype_of(str, AlwaysFalsy))
+
+static_assert(is_subtype_of(bool, Literal[False] | AlwaysTruthy))
+static_assert(is_subtype_of(bool, Literal[True] | AlwaysFalsy))
+static_assert(not is_subtype_of(Literal[True] | AlwaysFalsy, Literal[False] | AlwaysTruthy))
+static_assert(is_subtype_of(LiteralString, Literal[""] | AlwaysTruthy))
+
+# `LiteralString & AlwaysTruthy` -> `LiteralString & ~Literal[""]`
+static_assert(is_subtype_of(Intersection[LiteralString, AlwaysTruthy], AlwaysTruthy))
+static_assert(is_subtype_of(Intersection[LiteralString, Not[Literal[""]]], AlwaysTruthy))
+static_assert(is_subtype_of(Intersection[LiteralString, Not[Literal["", "a"]]], AlwaysTruthy))
+static_assert(not is_subtype_of(Intersection[object, Not[Literal[""]]], AlwaysTruthy))
+# `LiteralString & ~AlwaysTruthy` -> `LiteralString & Literal[""]`
+static_assert(is_subtype_of(Intersection[LiteralString, Not[AlwaysTruthy]], Not[AlwaysTruthy]))
+static_assert(is_subtype_of(Intersection[LiteralString, Literal[""]], Not[AlwaysTruthy]))
+static_assert(not is_subtype_of(Intersection[LiteralString, Literal["", "a"]], Not[AlwaysTruthy]))
+static_assert(is_subtype_of(Intersection[LiteralString, Not[AlwaysTruthy]], Literal[""]))
+# `bool & ~AlwaysTruthy`, `bool & ~Literal[True]` -> `bool & Literal[False]`
+static_assert(is_subtype_of(Intersection[bool, Not[AlwaysTruthy]], Literal[False]))
+static_assert(is_subtype_of(Intersection[bool, Not[Literal[True]]], Literal[False]))
```

Comment:
These three tests all look like they are describing a type equivalence, not just a subtype relationship. It seems they all pass if modified to use `is_equivalent_to` instead. Maybe we should also add these tests to `type_properties/is_equivalent_to.md`?